### ISSUE TITLE
Cannot end turn ("End Turn" button doesn't do anything)

### ISSUE THREAD

**Issue by poral** [LoneGazebo/Community-Patch-DLL#2505]
_1. Mod Version (i.e Date - 4/23):_
Vox Populi 9/12

_2. Mod List (if using Vox Populi only, leave blank):_

_3. Error Description:_ 
At some point in every one of my games so far, the "End Turn" button stops working. Reloading the save doesn't work. I experienced that issue with every version of Vox Populi (not with the versions before the CPP was renamed) so far.

_4. Steps to reproduce (optional):_

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):
[Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/468346/Logs.zip)

2. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

3. Screenshots (optional):

**Comment by LoneGazebo** [comment]
Interesting. @ilteroi, I've seen this issue once before with units set to explore not 'ending' their turn properly. Possible that we're back at this, again?

**Comment by LoneGazebo** [comment]
The lua log confirms this – something in the DLL is telling the notification system not to release the end turn, and I've seen this because of units not ending their turns properly.

@poral, try this: start again, but don't auto-explore any units.

**Comment by poral** [comment]
That works. The scout which was set on auto-explore had movement points left and therefore prevented me from advancing to the next turn. Thanks for your quick help!

**Comment by LoneGazebo** [comment]
Did the scout not use all its movement, yet his icon was greyed out? If so, that means it is not being properly 'finalized,' and that could be a problem.

**Comment by poral** [comment]
The scout did not move at all, and instead of auto-exploring he was just idling somewhere. The icon was not greyed out.

**Comment by LoneGazebo** [comment]
It was not greyed out? But still, it didn't prompt you to activate him? Hmm...

**Comment by poral** [comment]
![20160912232647_1](https://cloud.githubusercontent.com/assets/22152456/18453658/a96b293e-7940-11e6-900e-295f6d462464.jpg)

**Comment by LoneGazebo** [comment]
Interesting. That needs to be fixed. @ilteroi, I volunteer you as tribute :)

**Comment by ilteroi** [comment]
three scouts, thirty turns, no repro ... but just pushed another perf fix

**Comment by supermerill** [comment]
I've also seen this issue twice before. But i couldn't reproduced it reliably.

**Comment by Sekisetsu** [comment]
I can't end my turn in my current game too. but i have no scouts and no unit with auto explore or anything else. in this turn i just got a great engeneer and sent him where i need him and after this i can't end turn.

**Comment by LoneGazebo** [comment]
check all of your units, make sure they are actually finished on their turns.

**Comment by ilteroi** [comment]
@LoneGazebo : when you have a pending diplomacy notification, would it be possible to change the text in the next turn button?

**Comment by Iamblichos** [comment]
@ilteroi, event choices should be there if you are playing single player with modpack.

**Comment by LoneGazebo** [comment]
@ilteroi that is how it works now. There was simply a bug in the code during our test.

**Comment by ilteroi** [comment]
perfect. let's release this beast then!

**Comment by Iamblichos** [comment]
I gotta merge my code and then we test again.

**Comment by ilteroi** [comment]
another little thing .. sometimes capitals are being razed by the AI. i suspect it happens when a third party conquers the city. is that intended?

the relevant line in CanRaze() is:
```
	if(!bIgnoreCapitals && pCity->IsEverCapital() && bOriginalCapital)
               return false;
```
can't really make sense of it, what is the IsEverCapital() thing doing? it doesn't seem to be needed anywhere else.

**Comment by Iamblichos** [comment]
My code is merged...needs to be tested for sure.

**Comment by Iamblichos** [comment]
Just a little confirmation in modpack:

![20160916155608_1](https://cloud.githubusercontent.com/assets/14372214/18601030/4b065056-7c26-11e6-9beb-243d82074892.jpg)

**Comment by LoneGazebo** [comment]
Confirmation of working?

**Comment by Iamblichos** [comment]
Yes.

**Comment by Bloodrocution** [comment]
This issue is not fixed. I just experienced it today. I was able to get around it by using the In-Game Editor to delete the scout that was responsible for causing the bug. The scout was NOT on auto explore and did NOT have any movement points remaining. 

Aside from CBO (v 13), I the other mods I am using are 
1) More Luxuries
2) Faster Aircraft Animations (v3)
3) InfoAddict (v 22)
4) Ingame Editor (v 39.1)
5) Level Promotions (v 3) by Serp
6) Really Advanced Setup (v 15)

Of those, the only one I can see possibly causing a conflict is the Level Promotions mod. My scout did have a promotion for double movement speed on hills and he ended his turn on a hill. Seems inconsequential, but just throwing out any information, even if it's a stretch. I am running the latest version of the Community Patch (v 88).
